Switch to esm.sh source for handlebars package

mocoso opened the pull request:
Because dev.jspm.io has issues and installs are erroring

```
error: Import 'https://dev.jspm.io/npm:source-map@0.6?dew' failed: 500 Internal Server Error
    at https://dev.jspm.io/npm:handlebars@4.7.6/dist/cjs/handlebars/compiler/code-gen.dew.js:2:41
```

Note esm.sh's version of handlebars is only compatible with deno versions 1.24.0 and later and so this includes an upgrade of deno version for the tests

mocoso committed:
Switch to esm.sh source for handlebars package

Because dev.jspm.io has issues and installs are erroring

```
error: Import 'https://dev.jspm.io/npm:source-map@0.6?dew' failed: 500 Internal Server Error
    at https://dev.jspm.io/npm:handlebars@4.7.6/dist/cjs/handlebars/compiler/code-gen.dew.js:2:41
```

maplecfiv commented:
Refer to deno [v1.19.0](https://github.com/denoland/deno/releases/tag/v1.19.0) and PR[13660](https://github.com/denoland/deno/pull/13660)
Please consider upgrade the Deno version in test build

mocoso committed:
Upgrade deno to v1.24.0 to be compatible with esm.sh's handlebars

Earlier versions of deno report the following error when running the
tests

```
Check file:///home/runner/work/handlebars/handlebars/tests/basic.test.ts
error: TS2322 [ERROR]: Type 'AliasName<K, A>' is not assignable to type
'string | number | symbol'.
  Type 'K | (string extends A[K] ? K : A[K] extends string ? K | A[K] :
  K)' is not assignable to type 'string | number | symbol'.
      Type 'string extends A[K] ? K : A[K] extends string ? K | A[K] :
      K' is not assignable to type 'string | number | symbol'.
      ...
```

mocoso commented:
> Refer to deno [v1.19.0](https://github.com/denoland/deno/releases/tag/v1.19.0) and PR[13660](https://github.com/denoland/deno/pull/13660) Please consider upgrade the Deno version in test build

The esm.sh import still fails with v1.19.0.  with the error

```
Check file:///home/runner/work/handlebars/handlebars/tests/basic.test.ts
error: TS2322 [ERROR]: Type 'AliasName<K, A>' is not assignable to type 'string | number | symbol'.
  Type 'K | (string extends A[K] ? K : A[K] extends string ? K | A[K] : K)' is not assignable to type 'string | number | symbol'.
    Type 'string extends A[K] ? K : A[K] extends string ? K | A[K] : K' is not assignable to type 'string | number | symbol'.
      Type 'K | (A[K] extends string ? K | A[K] : K)' is not assignable to type 'string | number | symbol'.
        Type 'A[K] extends string ? K | A[K] : K' is not assignable to type 'string | number | symbol'.
          Type 'K | A[K]' is not assignable to type 'string | number | symbol'.
           ...
```

It only works with deno versions 1.24.0 and later

mocoso commented:
esm.sh looks like a better option than jspm which is proving unreliable even at it's new URL, see #16

irustm commented:
LGTM! Thanks!
